radicale_dovecot_auth/__init__.py

```diff
@@ -40,15 +40,14 @@ class Auth(BaseAuth):
 
     def get_connection(self):
         kwargs = {}
-        if self.configuration.has_option('auth', 'auth_socket'):
+        try:
             kwargs['socket_path'] = self.configuration.get('auth', 'auth_socket')
-
-        elif (self.configuration.has_option('auth', 'auth_host')
-              and self.configuration.has_option('auth', 'auth_port')):
             kwargs['host'] = self.configuration.get('auth', 'auth_host')
             kwargs['port'] = self.configuration.get('auth', 'auth_port')
+        except:
+            pass
 
-        else:
+        if not ('socket_path' in kwargs or {'host', 'port'} <= set(kwargs)):
```

Comment:
Why remove the explicit check? And I don't think I like that the try covers all three config lookups.

Reply:
The `Configuration` object is no longer a subclass of `ConfigParser`, so `has_options` no longer exists.